evil-common.el

```diff
@@ -1375,48 +1375,43 @@ match-data reflects the last successful match (that caused COUNT
 to reach zero). The behaviour of this functions is similar to
 `up-list'."
   (let* ((count (or count 1))
-         (dir (if (> count 0) +1 -1)))
+         (forwardp (> count 0))
+         (dir (if forwardp +1 -1))
+         (beg+ (if forwardp beg end))
```

Comment:
The `+` is rather misleading here, I'd have expected it to be an incremented beginning.  Perhaps it's best to replace the two references to each with the relevant if expression?